### PR TITLE
expose -untilItFails in hack/e2e.go

### PR DESCRIPTION
**Pull request by shiywang**
Fixes https://github.com/kubernetes/kubernetes/issues/43026

### CODE REVIEW

**Comment by spiffxp** [comment]
@k8s-bot ok to test

**Comment by fejta** [comment]
/unassign @eparis 
/assign

Thanks for the PR. Any chance that `go run hack/e2e.go -- --test --test_args=--ginkgo.untilItFails` accomplish the same thing?

**Comment by shiywang** [comment]
@fejta I try to run `go run hack/e2e.go -- --test --test_args=--ginkgo.untilItFails` seems failed due to `/home/centos/go/src/k8s.io/kubernetes/_output/bin/e2e.test` doesn't support this flag, same as jenkins failed log

**Comment by shiywang** [comment]
@fejta I check the the log, seems `e2e.test -h` doesn't show flag `ginkgo.untilItFails` , but I believe the flag is already in [here](https://github.com/kubernetes/kubernetes/blob/master/vendor/github.com/onsi/ginkgo/ginkgo/run_watch_and_build_command_flags.go#L153),  do I have to registry then using that flag ? but I see other flag like `ginkgo.failFast` seems doesn't have to

File: hack/ginkgo-e2e.sh [hunk 1]
```diff
@@ -104,6 +104,12 @@ elif [[ ${GINKGO_PARALLEL} =~ ^[yY]$ ]]; then
   ginkgo_args+=("--nodes=25")
 fi
 
+if [[ "${GINKGO_UNTIL_IT_FAILS}" == "true" ]]; then
```
Comment (fejta) [inline hack/ginkgo-e2e.sh]: `:-` so this does not blow up if the var is undefined.

File: hack/ginkgo-e2e.sh [hunk 2]
```diff
@@ -104,6 +104,12 @@ elif [[ ${GINKGO_PARALLEL} =~ ^[yY]$ ]]; then
...
+ginkgo_args+=("--untilItFails=true")
+else
```
Comment (fejta) [inline hack/ginkgo-e2e.sh]: Seems like the else part of this is unnecessary?

**Comment by shiywang** [comment]
@fejta updated, ptal

File: hack/ginkgo-e2e.sh [hunk 3]
```diff
@@ -104,6 +107,10 @@ elif [[ ${GINKGO_PARALLEL} =~ ^[yY]$ ]]; then
   ginkgo_args+=("--nodes=25")
 fi
 
+if [[ "${GINKGO_UNTIL_IT_FAILS:-}" == true ]]; then
+ginkgo_args+=("--untilItFails=true")
```
Comment (fejta) [inline hack/ginkgo-e2e.sh]: Cool! Have you confirmed that this works on your local machine?

File: hack/ginkgo-e2e.sh [hunk 4]
```diff
@@ -33,6 +33,9 @@ GINKGO_PARALLEL=${GINKGO_PARALLEL:-n} # set to 'y' to run tests in parallel
 # If 'y', will rerun failed tests once to give them a second chance.
 GINKGO_TOLERATE_FLAKES=${GINKGO_TOLERATE_FLAKES:-n}
 
+# If 'true', will run many times until it fails..
+UNTIL_IT_FAILS=false
```
Comment (fejta) [inline hack/ginkgo-e2e.sh]: I assume this is for testing purposes? And that you plan to remove this?

Comment (shiywang) [inline hack/ginkgo-e2e.sh]: removed

File: hack/ginkgo-e2e.sh (hunk 3 above)
Comment (shiywang) [inline hack/ginkgo-e2e.sh]: yes, with export env, it works well

**Comment by shiywang** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by shiywang** [comment]
@fejta  do you think we can merge this, after that I'll be working on moving ginkgo-e2e.sh scirpt to kubetest

**Comment by shiywang** [comment]
@k8s-bot cvm gce e2e test this

**Pull request merged**